src/Android/Avalonia.Android/AndroidInputMethod.cs

```diff
@@ -0,0 +1,95 @@
+using Avalonia.Input.TextInput;
+
+namespace Avalonia.Android
+{
+    class AndroidInputMethod<TView> : ITextInputMethodImpl
+        where TView: View, IInitEditorInfo
+    {
+        private readonly TView _host;
+        private readonly InputMethodManager _imm;
+        private IInputElement _inputElement;
+
+        public AndroidInputMethod(TView host)
+        {
+            if (host.OnCheckIsTextEditor() == false)
+                throw new InvalidOperationException("Host should return true from OnCheckIsTextEditor()");
+
+            _host = host;
+            _imm = host.Context.GetSystemService(Context.InputMethodService).JavaCast<InputMethodManager>();
+
+            _host.Focusable = true;
+            _host.FocusableInTouchMode = true;
+        }
+
+        public void Reset()
+        {
+            _imm.RestartInput(_host);
+        }
+
+        public void SetActive(bool active)
+        {
+            if (active)
+            {
+                _host.RequestFocus();
+                Reset();
+                _imm.ShowSoftInput(_host, ShowFlags.Implicit);
+            }
+            else
+                _imm.HideSoftInputFromWindow(_host.WindowToken, HideSoftInputFlags.None);
+        }
+
+        public void SetCursorRect(Rect rect)
+        {
+        }
+
+        public void SetOptions(TextInputOptionsQueryEventArgs options)
+        {
+            if (_inputElement != null)
+            {
+                _inputElement.PointerReleased -= RestoreSoftKeyboard;
+            }
+
+            _inputElement = options.Source as InputElement;
+
+            if (_inputElement == null)
+            {
+                _imm.HideSoftInputFromWindow(_host.WindowToken, HideSoftInputFlags.None);
+            }
+
+            _host.InitEditorInfo((outAttrs) =>
+            {
+                outAttrs.InputType = options.ContentType switch
+                {
+                    TextInputContentType.Email => global::Android.Text.InputTypes.TextVariationEmailAddress,
+                    TextInputContentType.Number => global::Android.Text.InputTypes.ClassNumber,
+                    TextInputContentType.Password => global::Android.Text.InputTypes.TextVariationPassword,
+                    TextInputContentType.Phone => global::Android.Text.InputTypes.ClassPhone,
+                    TextInputContentType.Url => global::Android.Text.InputTypes.TextVariationUri,
+                    _ => global::Android.Text.InputTypes.Null
+                };
+
+                if (options.AutoCapitalization)
+                {
+                    outAttrs.InitialCapsMode = global::Android.Text.CapitalizationMode.Sentences;
+                    outAttrs.InputType |= global::Android.Text.InputTypes.TextFlagCapSentences;
+                }
+
+                if (options.Multiline)
+                    outAttrs.InputType |= global::Android.Text.InputTypes.TextFlagMultiLine;
+            });
+
+            _inputElement.PointerReleased += RestoreSoftKeyboard;
```

Comment:
Why do you need to subscribe to PointerReleased? Input method management automatically tells you when new element needs input focus.

Reply:
When you press Back button Android hides keyboard, to show it again you do need tap input field (this is native Android behaviour). `ITextInputMethodImpl.SetActive()` is not called in this case (focus not changed). That's why I do nees subscribtion. `PointerReleased` is used because `PointerPressed` is handled already and not bypassed.

Reply:
How does the normal text view behave in that case? Does it keep the focus?

Reply:
Note, that we can't just assume which element is the actual text view

Reply:
> How does the normal text view behave in that case? Does it keep the focus?

can't say for now. I'll exam it with the real keyboard today.

Reply:
@kekekeks i'v checked few android applications, and in some input looks inactive when keyboard is hidden in other - disabled. So it is not clear if it accepts input.

Also soft keyboard is not shown at all when real keyboard is connected.

P.S. FUI tabulation works fine with real keyboard :)

Reply:
So we should probably just reset input focus when soft keyboard is dismissed

Reply:
I think it can be done on back button pressed handling, but this should be done in Avallnia's input framework

Reply:
@kekekeks  please review the latest changes (`SoftKeyboardListner`). I'll remove event subscribtion related code if this new approach would be consistent.